modules/drivers/snowflake/src/metabase/driver/snowflake.clj

```diff
@@ -229,12 +229,24 @@
 
 (defmethod sql.qp/->honeysql [:snowflake :datetime-diff]
   [driver [_ x y unit]]
-  (let [x (hsql/call :convert_timezone
-                     (qp.timezone/results-timezone-id)
-                     (sql.qp/->honeysql driver x))
-        y (hsql/call :convert_timezone
-                     (qp.timezone/results-timezone-id)
-                     (sql.qp/->honeysql driver y))]
+  (let [x (sql.qp/->honeysql driver x)
+        y (sql.qp/->honeysql driver y)
+        disallowed-types (keep
+                          (fn [v]
+                            (some-> v
+                                    hx/type-info
+                                    hx/type-info->db-type
+                                    name
+                                    str/lower-case
```

Comment:
You're going to have a really bad time if your system locale is Turkish with this code. Read the docstring for `metabase.util/lower-case-en` or give this a try in your REPL

```clj
(metabase.test/with-locale "tr" (clojure.string/lower-case "TIME"))
```

That's another argument for doing the error checking in one place instead of in every method implementation